File: src/main/java/com/scalag/vulkan/utility/VulkanObjectHandle.java
package com.scalag.vulkan.utility;

/**
 * @author MarconZet
 * Created 13.04.2020
 */
public abstract class VulkanObjectHandle extends VulkanObject {
    protected long handle;

    public long get(){
        if(!isAlive())
            throw new IllegalStateException();
        return handle;
    }
}

File: src/main/scala/com/scalag/vulkan/utility/VulkanObjectHandle.scala
package com.scalag.vulkan.utility;

/** @author
  *   MarconZet Created 13.04.2020
  */
abstract class VulkanObjectHandle extends VulkanObject {
  protected var handle: Long = 0L;

  def get: Long =
    if (!alive)
      throw new IllegalStateException();
    else
      handle;
}
